### PR TITLE
EG-3165: updating release notes for 4.5.1

### PR DESCRIPTION
**Pull request by AMcFarlandCampbellR3**

### CODE REVIEW

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 1]
```diff
@@ -16,6 +16,23 @@ weight: 1
...
+* Config Obfuscator doesn't work anymore for HSM configuration files
+* Retrying session inits can fail due to database connectivity
+* Revert H2 version to 197 to avoid bug introduced in the dependency after
+* NodeMeteringBackground processes high CPU while a node is idle
+* Prevent AMQP header spoofing 
+* Passing two sessions with the same counterparty to the CollectSignaturesFlow leads both counterparty's flows to infinitely await messages from other party
+* Uncaught exception in FlowStateMachineImpl.run().initialiseFlow()
+* [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE
+* TLS handshake timeout leads to blacklisting endpoint
+* Command line state ref usability improvement (CE 4.5.1)
```
Comment (benpesterr3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:35]:
```suggestion
* Classloader fails to find class when Optional generic used on Type definition.
* Config Obfuscator doesn't work anymore for HSM configuration files.
* Retrying session inits can fail due to database connectivity.
* Revert H2 version to 197 to avoid bug introduced in the dependency after
* NodeMeteringBackground processes high CPU while a node is idle.
* Prevent AMQP header spoofing.
* Passing two sessions with the same counterparty to the CollectSignaturesFlow leads both counterparty's flows to infinitely await messages from other party.
* Uncaught exception in FlowStateMachineImpl.run().initialiseFlow().
* [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE.
* TLS handshake timeout leads to blacklisting endpoint.
* Command line state ref usability improvement (CE 4.5.1).
```

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 2]
```diff
@@ -16,6 +16,23 @@ weight: 1
...
+## Corda Enterprise 4.5.1
+
+Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.
+
+### Issues Fixed
+
+* Classloader fails to find class when Optional generic used on Type definition
+* Config Obfuscator doesn't work anymore for HSM configuration files
+* Retrying session inits can fail due to database connectivity
+* Revert H2 version to 197 to avoid bug introduced in the dependency after
```
Comment (benpesterr3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:28]: Sentence seems to be incomplete -- after...

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 3]
```diff
@@ -16,6 +16,23 @@ weight: 1
 
 # Corda Enterprise release notes
 
+## Corda Enterprise 4.5.1
+
+Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.
+
+### Issues Fixed
```
Comment (benpesterr3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:23]: some of the issues are almost described as solutions. This is bit picky, but if you wanted to, you could change 'Command line state ref usability improvement' to something else? Or introduce the list as 

### Issues and solutions

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 2 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:28]: I'll follow this up with @fowlerrr - this line came directly from one of his JIRA tickets.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 3 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:23]: I think this is a bit picky, especially at the patch-release level. It is worth bearing in mind for future major releases though, and we should probably think about what we want to see in all release notes (including patch-level release notes) in Corda 5.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 2 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md]: Spoke to @fowlerrr on Slack and added "after previous upgrade." to the end of that bullet point.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 4]
```diff
@@ -16,6 +16,23 @@ weight: 1
...
+
+* Classloader fails to find class when Optional generic used on Type definition.
+* Config Obfuscator doesn't work anymore for HSM configuration files.
+* Retrying session inits can fail due to database connectivity.
+* Revert H2 version to 197 to avoid bug introduced in the dependency after previous upgrade.
```
Comment (VCAMP) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:28]: Actual version is 1.4.197, per `build.gradle` file

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 5]
```diff
@@ -16,6 +16,23 @@ weight: 1
 
 # Corda Enterprise release notes
 
+## Corda Enterprise 4.5.1
+
+Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.
+
+### Issues Fixed
+
+* Classloader fails to find class when Optional generic used on Type definition.
```
Comment (VCAMP) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:25]: Comments left on https://github.com/corda/corda-docs/pull/188#pullrequestreview-465350740 apply here too:

* Consistency of release note entries
* Entries containing square brackets

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 6]
```diff
@@ -16,6 +16,23 @@ weight: 1
...
+* Config Obfuscator doesn't work anymore for HSM configuration files.
+* Retrying session inits can fail due to database connectivity.
+* Revert H2 version to 197 to avoid bug introduced in the dependency after previous upgrade.
+* NodeMeteringBackground processes high CPU while a node is idle.
+* Prevent AMQP header spoofing.
+* Passing two sessions with the same counterparty to the CollectSignaturesFlow leads both counterparty's flows to infinitely await messages from other party.
+* Uncaught exception in FlowStateMachineImpl.run().initialiseFlow().
+* [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE.
+* TLS handshake timeout leads to blacklisting endpoint.
+* Command line state ref usability improvement (CE 4.5.1).
```
Comment (VCAMP) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:35]: Would rephrase the last point for clarity to:

"Added support to the spent state audit command for specifying state references in the form `txId(outputIdx)` in addition to the existing `txId:outputIdx`"

(from the ticket description)

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 3 above)
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:23]:
```suggestion
### Fixed issues
```

Comment (ivanterziev-r3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:23]: for consistency with CENM patch RNs etc. - although in future I would rather use "Resolved issues" instead, but maybe from Corda 5 onwards once we can cut the history line:)

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 5 above)
Comment (ivanterziev-r3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:25]: Same - please see https://github.com/corda/corda-docs/pull/188 and suggestions (for format/style) in https://github.com/corda/corda-docs/pull/190.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 7]
```diff
@@ -16,6 +16,23 @@ weight: 1
 
 # Corda Enterprise release notes
 
+## Corda Enterprise 4.5.1
+
+Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.
```
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:21]: Please add the upgrade recommendation here too. Although this IS the latest version now, it won't be once we release 4.6 and the upgrade boilerplate points to the actual current latest version, so I think it's good to have it in all patch release notes.

### Upgrade recommendation


As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 4 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:28]: Fixed

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 6 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:35]: Done

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 7 above)
Comment (AMcFarlandCampbellR3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:21]: Done.

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 8]
```diff
@@ -16,6 +16,29 @@ weight: 1
...
+
+As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).
+
+As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.
+
+### Fixed issues
+
+* Fixed an issue where the Classloader failed to find a Command class when Optional generic was used on Type definition.
+* The [Configuraton Obfuscator tool](tools-config-obfuscator.md) has been fixed to work for HSM configuration files.
+* Retrying session inits can fail due to database connectivity.
```
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:33]:
```suggestion
* Fixed an issue where retrying session inits could fail due to database connectivity.
```

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 9]
```diff
@@ -16,6 +16,29 @@ weight: 1
...
+* The H2 version has been reverted to 1.4.197 to avoid a dependency issue introduced after the previous upgrade.
+* The CPU usage of the `NodeMeteringBackground` process has been decreased.
+* A security update to prevent AMQP header spoofing has been applied.
+* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows to wait infinitely for messages from the other party.
```
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:37]:
```suggestion
* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows having to wait infinitely for messages from the other party.
```

Comment (maryforde) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md]: lead to both counterparties' flows having to wait infinitely for messages from the other party.

Needs to be "leads" or "led" above

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 10]
```diff
@@ -16,6 +16,29 @@ weight: 1
...
+
+### Fixed issues
+
+* Fixed an issue where the Classloader failed to find a Command class when Optional generic was used on Type definition.
+* The [Configuraton Obfuscator tool](tools-config-obfuscator.md) has been fixed to work for HSM configuration files.
+* Fixed an issue where retrying session inits could fail due to database connectivity.
+* The H2 version has been reverted to 1.4.197 to avoid a dependency issue introduced after the previous upgrade.
+* The CPU usage of the `NodeMeteringBackground` process has been decreased.
+* A security update to prevent AMQP header spoofing has been applied.
+* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows having to wait infinitely for messages from the other party.
```
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:37]:
```suggestion
* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` led to both counterparties' flows having to wait infinitely for messages from the other party.
```

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md [hunk 11]
```diff
@@ -16,6 +16,29 @@ weight: 1
...
+* Fixed an issue where the Classloader failed to find a Command class when Optional generic was used on Type definition.
+* The [Configuraton Obfuscator tool](tools-config-obfuscator.md) has been fixed to work for HSM configuration files.
+* Fixed an issue where retrying session inits could fail due to database connectivity.
+* The H2 version has been reverted to 1.4.197 to avoid a dependency issue introduced after the previous upgrade.
+* The CPU usage of the `NodeMeteringBackground` process has been decreased.
+* A security update to prevent AMQP header spoofing has been applied.
+* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` led to both counterparties' flows having to wait infinitely for messages from the other party.
+* A previously unhandled exception in `FlowStateMachineImpl.run().initialiseFlow()` is now handled correctly.
+* Fixed an issue where Corda Firewall did not start if its main configuration and its HSM configuration were obfuscated.
+* Fixed an issue where a TLS handshake timeout lead to blacklisting endpoint.
```
Comment (ivanterziev-r3) [inline content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md:40]:
```suggestion
* Fixed an issue where a TLS handshake timeout led to blacklisting endpoint.
```

File: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md (hunk 9 above)
Comment (ivanterziev-r3) [reply content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md]: Yep well-spotted! Fixed in two places.
